### ISSUE TITLE
Can't resolve all dependencies

### ISSUE THREAD

**Issue by gultyaev** [angular/angular-cli#15917]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/cli
Probably by @angular-devkit/build-angular

### Is this a regression?

 Yes, the previous version in which this bug was not present was: ....
It was fine when I was working on `@angular/cli: 8.3.6` and 
`@angular-devkit/build-angular: 0.802.2`

### Description

 A clear and concise description of the problem...
The problem arises when Class A extends Class B which has a constructor and Class A doesn't specify this constructor.

Before the update it was working and after I updated my dependencies it stopped. It started alerting that "cannot resolve all dependencies".

After I specified a constructor with super call explicitly the error vanished.

## 🔬 Minimal Reproduction

 https://stackblitz.com/edit/angular-issue-repro2-tncgci

## 🔥 Exception or Error
<pre><code>

Can't resolve all parameters for AppComponent: (?).
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 8.3.9
Node: 12.11.1
OS: linux x64
Angular: 8.2.10
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.803.9
@angular-devkit/build-angular      0.803.9
@angular-devkit/build-ng-packagr   0.803.9
@angular-devkit/build-optimizer    0.803.9
@angular-devkit/build-webpack      0.803.9
@angular-devkit/core               8.3.9
@angular-devkit/schematics         8.3.9
@angular/cdk                       8.2.3
@angular/cli                       8.3.9
@angular/material                  8.2.3
@angular/material-moment-adapter   8.2.3
@ngtools/webpack                   8.3.9
@schematics/angular                8.3.9
@schematics/update                 0.803.9
ng-packagr                         5.6.1
rxjs                               6.5.3
typescript                         3.4.5
webpack                            4.39.2

</code></pre>

**Anything else relevant?**

**Comment by alan-agius4** [comment]
Most likely this has the same resolution of https://github.com/angular/angular-cli/issues/15892

**Comment by AnandkumarBShetty** [comment]
> Most likely this has the same resolution and cause of #15892

Looked at the #15892. My code has export class EffectsName(not a component). When added the directive before the export as suggested in #15892, like @Directive(), this is not working.  If required i can show my code to you on any direct chat.

**Comment by alan-agius4** [comment]
@AnandkumarBShetty, it would be great to share the code or create a minimal reproduction. A good way to make a minimal repro is to create a new app via `ng new repro-app` and adding the minimum possible code to show the problem. Then you can push this repository to github and link it here.

**Comment by antch** [comment]
@AnandkumarBShetty I am running into the same issue when upgrading from `@angular-devkit/build-angular: 0.802.2` -- did you ever find a workaround?

**Comment by antch** [comment]
By decorating the base class with `@Injectable()`, it works:
https://stackblitz.com/edit/angular-issue-repro2-9kqik4

**Comment by AnandkumarBShetty** [comment]
> By decorating the base class with `@Injectable()`, it works:
> https://stackblitz.com/edit/angular-issue-repro2-9kqik4
> 
> The issue that I am running into is a little different, I have a class which gets deps via constructor, and is injected only via provider. By decorating this class with `@Injectable()` I was able to get around the error. The weird thing is that it was working fine in prod mode.

Excellent!. Thank you for this. The issue got resolved after decorating the @Injectable() for every Effects defined in the same file. Though only once the @Injectable() for one of the effect, it used to run into issue for other effects which was not clear why. Your input helped to resolve this long standing issue. Once again thank you for helping with your comment on this issue despite the issue being in dormant state for quite sometime.

**Comment by antch** [comment]
No problem. I will chime in by saying that even though the workaround is fairly straight-forward it doesn't really make sense. Why would I have to make an `abstract` class `@Injectable()`? It's never going to be injected anywhere.

I also suspect, given the low complexity of my apps and that both encountered this issue, that this is going to break many, many apps once they upgrade. I had to do so due to security vulnerability reported by `npm audit`.

**Comment by MikeKovetsky** [comment]
Was resolved by @angular-devkit/build-angular downgrade. The features of @angular-devkit/build-angular: 0.803.* broke the backward compatibility in JIT. Please, make the versioning clearer.

**Comment by alan-agius4** [comment]
Hi, this issue is now obsolete in version 9+, because base classes that use DI need to be decorated.

When using `ng update` to update to Angular version 9, base classes will be decorated via a migration. See https://angular.io/guide/migration-undecorated-classes for information.

Please update to the most recent Angular CLI version and If the problem persists after upgrading, please open a new issue, provide a simple repository reproducing the problem, and describe the difference between the expected and current behavior.
